py-polars/polars/utils/_construction.py

```diff
@@ -1443,9 +1443,29 @@ def arrow_to_pydf(
     rechunk: bool = True,
 ) -> PyDataFrame:
     """Construct a PyDataFrame from an Arrow Table."""
+
+    # Polars requires unique column names. In case the arrow table contains
+    # duplicate column names, the schema must be passed as it is used to
+    # override the column names of the arrow table.
+    if schema is None:
+        used_names = set()
+
+        for name in data.column_names:
+            if name in used_names:
+                raise ValueError(
+                    f"Arrow table contained duplicate column '{name}'. Please "
+                    "rename the columns to remove duplicates or pass a schema "
+                    "to be used to override the column names. Note that the "
+                    "schema must be in the form of a dictionary if a column "
+                    "uses an empty string '' as its name."
+                )
+            used_names.add(name)
+
     original_schema = schema
+    # Call _unpack_schema with a dictionary to preserve empty string ''.
     column_names, schema_overrides = _unpack_schema(
-        (schema or data.column_names), schema_overrides=schema_overrides
+        (schema or {x: None for x in data.column_names}),  # type: ignore[misc]
```

Comment:
Not sure why mypy complains here - `polars/utils/_construction.py:1467: error: Value expression in dictionary comprehension has incompatible type "None"`; expected type "DataTypeClass | DataType | (...) | type[Decimal] | type[None]", but `type[None]` is also an expected type.